### PR TITLE
fix #2356 Have Direct/EmitterProcessor implement currentContext()

### LINKED ISSUE #2356: DirectProcessor and EmitterProcessor don't override currentContext()
## Expected Behavior
Calling `currentContext()` on a `DirectProcessor` or `EmitterProcessor` with a single subscriber that _has_ a `Context` should return that context.

## Actual Behavior
`Context.empty()` is always returned.

## Possible Solution
Override the method and use `Operators.multiSubscribersContext(subscribers);` (like in `ReplayProcessor`).

### PR DESCRIPTION
**Pull request by simonbasle**
this is targeting 3.3.10.RELEASE and to be forward-merged into 3.4.0-RC1

### CODE REVIEW

**Comment by bsideup** [comment]
@simonbasle could you please check why ArchUnit tests did not detect this issue?

**Review by aneveu — APPROVED** [review]

**Comment by simonbasle** [comment]
@bsideup totally forgot about the currentContext ArchUnit test. It looks for `CoreSubscriber` derivatives, not `FluxProcessor` ones.

File: reactor-core/src/test/java/reactor/core/publisher/ProcessorArchTest.java [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+import com.tngtech.archunit.core.importer.ClassFileImporter;
+import com.tngtech.archunit.core.importer.ImportOption;
+import com.tngtech.archunit.lang.ArchCondition;
+import com.tngtech.archunit.lang.ConditionEvents;
+import com.tngtech.archunit.lang.SimpleConditionEvent;
+import org.junit.Test;
+
+import static com.tngtech.archunit.lang.syntax.ArchRuleDefinition.classes;
+
+public class ProcessorArchTest {
```
Comment (bsideup) [inline reactor-core/src/test/java/reactor/core/publisher/ProcessorArchTest.java:31]: can't we just adjust the current one? e.g. merge the classes plus do a selective `anyMatch`

Comment (simonbasle) [reply reactor-core/src/test/java/reactor/core/publisher/ProcessorArchTest.java:31]: yeah we could. the archunit class was named `CoreSubscriberArchTest` and resides in a different package though... But I could maybe refocus the archunit test on `currentContext` in general by renaming it to `CurrentContextArchTest`?

**Review by bsideup — APPROVED** [review]

**Pull request merged**